Register PassFast & FailFast ends from their respective DSL options

yogeshjain999 opened the pull request:
#### Issue

```ruby
# Below step raises exception "Unrecognized Signal `Trailblazer::Activity::FastTrack::PassFast`"
# even though `pass_fast` option is passed.
step task: ->((ctx, flow_options), **) { return PassFast, [ctx, flow_options] }, pass_fast: true
```

#### Fix
Add `PassFast` or `FailFast`end when it's respective option is given.

It also fixes breaking test reported in https://github.com/trailblazer/trailblazer-macro/commit/7ff85f551ac1ae9b25b333b1f998d5c572ce415f

apotonick reviewed (commented):
I don't think we are going to have side-effects here? Or could the behavior change with `Subprocess`'es nesting?

yogeshjain999 commented:
Nope, there won't be any side effects, just adding more `ends` on the step itself. But I updated test by adding 2 levels of nesting on the safe side.

Also added an `assert_raises` test where if nested `Subprocess` emits any fast track signal but parent hasn't registered for it.
